File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1

    clothesType = dict()

    for each in clothes:
        if each[1] not in clothesType:
            clothesType[each[1]] = 1
        else:
            clothesType[each[1]] += 1

    
    for val in clothesType.values():
        answer *= (val + 1)

    answer -= 1
    
    return answer

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> types = new HashMap<>();
        
        for (String[] cloth : clothes) {
            String type = cloth[1];
            String name = cloth[0];
            
            types.put(type, types.getOrDefault(type, 0) + 1);
        }
        
        for (int n : types.values()) {
            answer *= (n + 1);
        }
        
        return answer - 1;
    }
}